src/app/config/or_state_st.h

```diff
@@ -87,6 +87,18 @@ struct or_state_t {
 
   /** When did we last rotate our onion key?  "0" for 'no idea'. */
   time_t LastRotatedOnionKey;
+
+  /** Do we automatically fail IPv4 or IPv6? If so, set one of these to 1. */
+  int IPv4AutoFail;
```

Comment:
This name is confusing:
* the variable actually tracks if we have ever failed an IPv4 connection because there was no route to a host.
* but maybe it should track the number of failed connections due to no route to host.